models/content.go

```diff
@@ -74,3 +67,51 @@ type ContentLinks struct {
 }
 
 // !!! add code to validate state transitions as per topic.go
+//!!! fix the following, and sort test code elsewhere, as per topic
+
+// Validate checks that a content struct complies with the state constraints, if provided. !!! may want to add more in future
+func (t *Content) Validate() error {
+
+	if _, err := ParseState(t.State); err != nil {
+		return apierrors.ErrTopicInvalidState
+	}
+
+	// !!! add other checks, etc
+	return nil
+}
+
+// ValidateTransitionFrom checks that this content state can be validly transitioned from the existing state
+func (t *Content) ValidateTransitionFrom(existing *Content) error {
+
+	// check that state transition is allowed, only if state is provided
+	if t.State != "" {
+		if !existing.StateTransitionAllowed(t.State) {
+			return apierrors.ErrTopicStateTransitionNotAllowed
+		}
+	}
+
+	// if the topic is already completed, it cannot be updated
```

Comment:
Remove commented out code.

If all the comments in here relate to long term reminders (for when we extend the system for writes in the future) it would probably be best to prefix them with the more standard TODO rather than !!! so tooling can pick it up. If they're short term reminders for your next ticket or so then they can stay as is, if they'll be removed after the work is done

Reply:
removed